tests/test_kubernetes_master.py

```diff
@@ -121,3 +121,23 @@ def test_update_certificates_with_missing_relations(mock_send_data,
     hookenv.log.assert_any_call("Missing relations: 'certificates.available, "
                                 "kube-api-endpoint.available'", hookenv.ERROR)
     mock_send_data.assert_not_called()
+
+
+@mock.patch("reactive.kubernetes_master.is_flag_set")
+@mock.patch("reactive.kubernetes_master.hookenv.status_set")
+def test_status_set_on_missing_ca(mock_status_set, mock_is_flag_set):
+    """Test that set_final_status() will set blocked state if CA is missing"""
+    def is_flag_set_side_effect(flag):
+        flags = {'upgrade.series.in-progress': False,
+                 'certificates.available': False}
+        value = flags.get(flag, None)
+        assert value is not None, "Unexpected flag requested: {}".format(flag)
+        return value
+
+    mock_is_flag_set.side_effect = is_flag_set_side_effect
+
+    kubernetes_master.set_final_status()
+
+    mock_status_set.assert_called_once_with('blocked',
+                                            'Missing relation to certificate '
+                                            'authority.')
```

Comment:
These charms use [charms.unit_test](https://github.com/juju-solutions/charms.unit_test/) which, while poorly documented, [patches out the flag helpers](https://github.com/juju-solutions/charms.unit_test/blob/master/charms/unit_test.py#L269-L272) with fakes that should work the same, so this can be changed to:

```suggestion
def test_status_set_on_missing_ca():
    """Test that set_final_status() will set blocked state if CA is missing"""

    set_flag("certificates.available")
    set_flag("kubernetes-master.secure-storage.failed")

    kubernetes_master.set_final_status()
    hookenv.status_set.assert_called_with('blocked',
                                          'Failed to configure encryption; '
                                          'secrets are unencrypted or inaccessible')

    clear_flag("certificates.available")
    kubernetes_master.set_final_status()
    hookenv.status_set.assert_called_with('blocked',
                                          'Missing relation to certificate '
                                          'authority.')
```

Arguable if that's any clearer or not, but it would be a bit more consistent with the other tests. Mostly I'm just putting it out there and am +1 to merging as-is.